src/main/java/query/QueryExecution.java

```diff
@@ -0,0 +1,211 @@
+import org.opensearch.sql.executor.ExecutionEngine.QueryResponse;
+import org.opensearch.sql.executor.ExecutionEngine.Schema;
+import org.opensearch.sql.ppl.PPLService;
+import org.opensearch.sql.ppl.domain.PPLQueryRequest;
+import org.opensearch.sql.protocol.response.QueryResult;
+import org.opensearch.sql.protocol.response.format.CsvResponseFormatter;
+import org.opensearch.sql.protocol.response.format.JdbcResponseFormatter;
+import org.opensearch.sql.protocol.response.format.JsonResponseFormatter;
+import org.opensearch.sql.protocol.response.format.RawResponseFormatter;
+import org.opensearch.sql.protocol.response.format.ResponseFormatter;
+import org.opensearch.sql.protocol.response.format.SimpleJsonResponseFormatter;
+import org.opensearch.sql.sql.SQLService;
+import org.opensearch.sql.sql.domain.SQLQueryRequest;
+
+public class QueryExecution {
+
+  private final PPLService pplService;
+  private final SQLService sqlService;
+
+  @Inject
+  public QueryExecution(PPLService pplService, SQLService sqlService) {
+    this.pplService = pplService;
+    this.sqlService = sqlService;
+  }
+
+  public String execute(String query, boolean isPPL) {
+    return execute(query, isPPL, "json");
+  }
+
+  public String execute(String query, boolean isPPL, String format) {
+    System.out.println("Received query: " + query);
+    System.out.println("Query type: " + (isPPL ? "PPL" : "SQL"));
+
+    try {
+      CountDownLatch latch = new CountDownLatch(1);
+      AtomicReference<QueryResponse> executeRef = new AtomicReference<>();
+      AtomicReference<Exception> errorRef = new AtomicReference<>();
+      AtomicReference<ExplainResponse> explainRef = new AtomicReference<>();
+      ResponseListener<QueryResponse> queryListener =
+          new ResponseListener<>() {
+            @Override
+            public void onResponse(QueryResponse response) {
+              System.out.println("Execute Result: " + response);
+              executeRef.set(response);
+              latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+              System.out.println("queryExecution Execution Error: " + e);
+              errorRef.set(e);
+              latch.countDown();
+            }
+          };
+
+      ResponseListener<ExplainResponse> explainListener =
+          new ResponseListener<>() {
+            @Override
+            public void onResponse(ExplainResponse response) {
+              System.out.println("Explain response: " + response);
+              explainRef.set(response);
+              latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+              System.out.println("queryExecution Explain Error: " + e);
+              errorRef.set(e);
+              latch.countDown();
+            }
+          };
+
+      // Check if this is an explain query
+      boolean isExplainQuery = query.trim().toLowerCase().startsWith("explain");
```

Comment:
why hardcode this instead of using parser? https://github.com/opensearch-project/sql/blob/e23686e58ef1d7d759342669d8b906c94b7253fa/ppl/src/main/antlr/OpenSearchPPLParser.g4#L26